samples/drivers/gpio_bd8lb600fs/README.rst

```diff
@@ -0,0 +1,50 @@
+.. _gpio_bd8lb600fs_sample:
```

Comment:
Hmm... this is interesting. So normally, I would advise against a driver-implementation specific sample, but of course this is an interesting case where really any board with an exposed SPI port could be used to test this SPI GPIO expander.

Really, we do need to add entries for this driver to:
* `tests/drivers/build_all/gpio/`
* `tests/drivers/gpio/gpio_api_1pin/`
* `tests/drivers/gpio/gpio_basic_api/`

Since the peripheral is not hard-wired to the `nucleo_f429zi` board, it might make sense to include a `gpio_bd8lb600fs.overlay` and a `gpio_bd8lb600fs.conf` in the test directories, and to include additional `extra_configs` or `extra_args` as necessary to a new entry in `testcase.yaml`.

Still buildable with west as well, using the same `extra_configs` or `extra_args`.

Now - the fact that there is a state to restore (presumably this is w.r.t. power management), might need to be handled in a separate way, but lets get to that bridge after the next rev.

Reply:
I will try to figure this out and add the necessary entries. The reason for the sample was that I needed to test it anyway, hence I thought I could just submit my test project as well. It will probably make it easier for somebody else to use this driver as well, as there is a template on how it can be configured.

Reply:
Definitely would be helpful to others - no argument there.

The main issue is we have to try to keep samples to a minimum just because there are just so many unique devices and boards. 

At the same time, we want all GPIO devices to look and feel approximately the same.

With that, generic tests (and device-specific `.overlay` and `.conf`) should be enough to get people off the ground.

Reply:
I did not add this specific driver to gpio_basic_api, as it can only be used as an output. Otherwise I went ahead and tried to add this, hopefully correctly.